### DISCUSSION TITLE
Gaussian Mixture Model in R Studio

### DISCUSSION THREAD

**Discussion by yohanespm77** [teamalgoritma/community#648] [Q&A] [answered]
Saya ingin melakukan outlier detection menggunakan model GMM, kodenya seperti berikut : 

<img width="707" alt="image" src="https://user-images.githubusercontent.com/106010284/235639501-eba1e974-d8ae-40d7-a698-9ee82f7e2ed6.png">

Namun hasilnya malah semua titik terdeteksi sebagai outlier :
<img width="529" alt="image" src="https://user-images.githubusercontent.com/106010284/235639606-f27bd59f-39e5-4abe-a95d-360dde142449.png">

Apabila saya mengganti nilai G = 3, malah terjadi error : 

<img width="649" alt="image" src="https://user-images.githubusercontent.com/106010284/235639703-6686fc67-a756-48c5-a72a-2a9bc5d0f2a8.png">

Mohon bantuannya kak, terima kasih 🙏

**Comment by idoali** [comment, marked as answer]
Terima kasih pak atas pertanyaannya

Error sendiri disebabkan karena nilai `G` itu menunjukkan jumlah cluster. Jika jumlah cluster tidak sesuai dengan kondisi data maka akan menyebabkan error. Sedangkan pada gambar kedua, dimana pada gambar tersebut diperlihatkan outlier yang terjadi hampir di semua titik, mungkin ini disebabkan oleh data bapak yang terlalu random. Mungkin bisa dicoba untuk diganti terlebih dulu pak datanya dengan data baru yang tidak terlalu random.

Semoga dapat membantu pak
